[TP] Add deprecation warnings in the documentations for Pairwise parallel, sequence parallel and other prepare input/output functions

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #111177
* __->__ #111176
* #111166
* #111160

As part of TP UX improvements, we want to keep our API simple (not easy) so that users get the flexibility to do what they want and avoid a too generic API which tries to solve everything and get things too complicated. We are updating the doc accordingly.

fduwjj committed:
Remove documentations for Pairwise parallel and sequence parallel

[ghstack-poisoned]

wanchaol commented on docs/source/distributed.tensor.parallel.rst:
```diff
@@ -27,31 +27,20 @@ Tensor Parallelism supports the following parallel styles:
-used for input/output preparation:
+DTensor layout of the input and output of the module so it can interact with
+the module parameters and module afterwards. Users can achieve this by specifying
+the ``input_layouts`` and ``output_layouts`` which decide what the DTensor of inputs
```
```suggestion
the ``input_layouts`` and ``output_layouts`` which annotate inputs to DTensors and redistribute the outputs if needed.
```

wanchaol commented on docs/source/distributed.tensor.parallel.rst:
```diff
@@ -27,31 +27,20 @@ Tensor Parallelism supports the following parallel styles:
+and outputs.
 
+If users only want to annotate the DTensor layout for inputs/outputs and no need to
+distribute its parameters, the following classes can be used in the ``parallelize_plan``
```
```suggestion
distribute its parameters, the following styles can be used in the ``parallelize_plan``
```

fduwjj replied:
But the following are actually styles?

wanchaol commented on docs/source/distributed.tensor.parallel.rst:
```diff
@@ -27,31 +27,20 @@ Tensor Parallelism supports the following parallel styles:
+.. autofunction::  PrepareModuleInput
+.. autofunction::  PrepareModuleOutput
 
 Currently, there are some constraints which makes it hard for the ``MultiheadAttention``
```
Let's update this part of doc too:


Currently we recommend users to use ``ColwiseParallel``
and ``RowwiseParallel`` for each Linear/Embeeding layers to compose Attention Parallelization. There might be some code changes needed since we are parallelizing on the head dim of the ``MultiheadAttention`` module.

fduwjj replied:
sure.

fduwjj replied:
Maybe I should do this in the next PR.

wanchaol reviewed (approved):
looks good

wanchaol commented on docs/source/distributed.tensor.parallel.rst:
```diff
@@ -27,31 +27,20 @@ Tensor Parallelism supports the following parallel styles:
 .. autoclass:: torch.distributed.tensor.parallel.style.ColwiseParallel
   :members:
```
We should also make sure ColwiseParallel and RowwiseParallel doc (i.e. __init__ doc) with the input_layouts, output_layouts, etc. appear in the doc

fduwjj replied:
I will add in the next PR.

fduwjj committed:
Update on "[TP] Remove documentations for Pairwise parallel, sequence parallel and other prepare input/output functions"


As part of TP UX improvements, we want to keep our API simple (not easy) so that users get the flexibility to do what they want and avoid a too generic API which tries to solve everything and get things too complicated. We are updating the doc accordingly.


[ghstack-poisoned]

fduwjj committed:
Update on "[TP] Remove documentations for Pairwise parallel, sequence parallel and other prepare input/output functions"


As part of TP UX improvements, we want to keep our API simple (not easy) so that users get the flexibility to do what they want and avoid a too generic API which tries to solve everything and get things too complicated. We are updating the doc accordingly.


[ghstack-poisoned]

fduwjj committed:
Update on "[TP] Remove documentations for Pairwise parallel, sequence parallel and other prepare input/output functions"


As part of TP UX improvements, we want to keep our API simple (not easy) so that users get the flexibility to do what they want and avoid a too generic API which tries to solve everything and get things too complicated. We are updating the doc accordingly.


[ghstack-poisoned]

wanchaol commented:
https://docs-preview.pytorch.org/pytorch/pytorch/111176/distributed.tensor.parallel.html looks like prepare input/outout docs still have issues

fduwjj commented:
@pytorchbot merge

fduwjj commented:
@wanchaol Let me add warnings for them in https://github.com/pytorch/pytorch/pull/111177.

wanchaol commented:
@fduwjj This is not related to deprecation warning, see here https://docs-preview.pytorch.org/pytorch/pytorch/111176/distributed.tensor.parallel.html#torch.distributed.tensor.parallel.style.PrepareModuleOutput

The examples in the doc are not properly formatted
